x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/accesscontrol/FieldSubsetReader.java

```diff
@@ -195,6 +195,8 @@ public Fields getTermVectors(int docID) throws IOException {
                 Map<String, Object> filteredValue = filter((Map<String, ?>)value, includeAutomaton, state);
                 if (filteredValue.isEmpty() == false) {
                     filtered.add(filteredValue);
+                } else {
+                    filtered.add(new HashMap<>());
```

Comment:
Maybe use the empty filteredValue directly ?

Reply:
Good point. I've updated the PR.